packages/core/src/wallet/IndyWallet.ts

```diff
@@ -129,9 +157,13 @@ export class IndyWallet implements Wallet {
     try {
       this.walletHandle = await this.indy.openWallet(
         { id: walletConfig.id },
-        { key: walletConfig.key, key_derivation_method: walletConfig.keyDerivationMethod }
+        { key: walletConfig.key, rekey: rekey, key_derivation_method: walletConfig.keyDerivationMethod }
       )
-      this.walletConfig = walletConfig
+      if (this.walletConfig) {
+        this.walletConfig = { ...this.walletConfig, ...walletConfig }
+      } else {
+        this.walletConfig = walletConfig
+      }
```

Comment:
If rekey is set we can maybe set the wallet config with the new key as the key:

```suggestion
      if (walletConfig.rekey) {
        const { rekey, key, keyDerivationMethod, rekeyDerivationMethod, ...rest } = walletConfig
        this.walletConfig = { ...rest, key: rekey, keyDerivationMethod: rekeyDerivationMethod }
      } else {
        this.walletConfig = walletConfig
      }
```